Sources/PrinceOfVersions/PoVDataTypes/PoVRequestOptions.swift

```diff
@@ -31,16 +31,39 @@ public class PoVRequestOptions: NSObject {
     // MARK: - Public methods
 
     /**
-    Adds requirement check for configuration.
+     Adds requirement check for configuration.
 
      Use this method to add custom requirement by which configuration must comply with.
 
      - parameter key: String that matches key in requirements array in JSON with `requirementsCheck` parameter,
-     - parameter requirementCheck: A block used to check if a configuration meets requirement. This block returns `true` if configuration meets the requirement, and takes the any value as input parameter by which .
+     - parameter type: The expected type of the value.
+     - parameter requirementCheck: A block used to check if a configuration meets the requirement. This block returns `true` if the configuration meets the requirement, and takes the typed value as input.
 
      */
-    public func addRequirement(key: String, requirementCheck: @escaping ((Any) -> Bool)) {
-        userRequirements.updateValue(requirementCheck, forKey: key)
+    public func addRequirement<T>(key: String, ofType type: T.Type, requirementCheck: @escaping (T) -> Bool) {
+        userRequirements.updateValue({ value in
+            guard let typedValue = value as? T else { return false }
+            return requirementCheck(typedValue)
+        }, forKey: key)
+    }
+
+    /**
+     Adds requirement check for configuration (Objective-C compatible).
+
+     Use this method to add a custom requirement by which configuration must comply with.
+
+     - parameter key: String that matches the key in the requirements array in JSON with the `requirementCheck` parameter.
+     - parameter type: The expected class of the value (e.g., `NSString.class`).
+     - parameter requirementCheck: A block used to check if a configuration meets the requirement. This block returns `true` if the configuration meets the requirement, and takes the value as input.
+
+     This method is designed for Objective-C compatibility and uses runtime type checking (`isKindOfClass:`) to validate the value.
+     */
+    @objc(addRequirementWithKey:ofType:requirementCheck:)
+    public func addRequirementWithKey(key: String, ofType type: AnyClass, requirementCheck: @escaping (Any) -> Bool) {
```

Comment:
Can we also add unavailable for swift flag, the last time I did it, i think it was adding __ to the function name, but there might be be a better option now.